clientd/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+use fedimint_client::{Client, UserClientConfig};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use serde_json::json;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use strum::EnumIter;
+use strum::IntoEnumIterator;
+use tokio::sync::{broadcast, mpsc, oneshot};
+
+mod rpc;
+pub mod server;
+
+// TODO: remove pub and use this type to enforce invariants
+#[derive(Debug, Serialize)]
+pub struct CallbackResponse(pub Result<String, RpcError>);
+
+// TODO: figure out best approach for error handling
+#[derive(Debug, Serialize)]
+pub enum RpcError {
+    ClientError,
+}
+#[derive(Debug)]
+pub enum ManagerMessage {
+    CallHandler(RpcRequest, oneshot::Sender<CallbackResponse>),
+    HandleEvent(Event),
+}
+
+#[derive(Debug)]
+pub enum RpcRequest {
+    HealthCheck(),
+    Info(),
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct Event {
+    event_key: EventKey,
+    data: String,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, EnumIter, Serialize)]
+pub enum EventKey {
+    Mock,
+}
+pub async fn run_manager(
+    _manager_tx: Arc<mpsc::Sender<ManagerMessage>>, // we will give the manager_tx to worker-tasks that can emit events
+    mut manager_rx: mpsc::Receiver<ManagerMessage>,
+    mut subscribers: HashMap<EventKey, broadcast::Sender<Event>>,
+    mut fedimint_client: Client<UserClientConfig>,
+) {
+    while let Some(msg) = manager_rx.recv().await {
+        match msg {
+            ManagerMessage::CallHandler(request, callback) => {
+                let response = handle_rpc_request(request, &mut fedimint_client).await;
+                callback
+                    .send(CallbackResponse(response))
+                    .expect("the rpc-method closure did not drop the callback receiver");
+            }
+            ManagerMessage::HandleEvent(event) => {
+                // the tx is broadcast so this will handle all the subscribers
+                let subscriber = subscribers.get_mut(&event.event_key).expect(
+                    "if the event exists it was inserted into the hashmap at initialization",
+                );
+                if subscriber.send(event).is_err() {
+                    // we will do if let Err(e) => handle error later but for now make the linter happy
+                    tracing::error!("error broadcasting the event to the subscribers");
+                }
+            }
+        }
+    }
+}
+
+pub fn map_subscribers_to_events() -> HashMap<EventKey, broadcast::Sender<Event>> {
```

Comment:
`HashMap<EventKey, broadcast::Sender<Event>>` should probably become a newtype with this as its `new` function.